examples/vortex-mpi.py

```diff
@@ -183,14 +185,22 @@ def my_checkpoint(step, t, dt, state):
                               exittol=exittol, constant_cfl=constant_cfl, comm=comm,
                               vis_timer=vis_timer)
 
+    def my_simhealthcheck(state, step, t, dt):
+        cv = split_conserved(discr.dim, state)
+        return sim_healthcheck(discr, eos, q=state,
+                               conserved_vars=cv,
+                               step=step, t=t)
+
     try:
         (current_step, current_t, current_state) = \
             advance_state(rhs=my_rhs, timestepper=timestepper,
-                checkpoint=my_checkpoint,
```

Comment:
These examples need to keep `my_checkpoint` as the call-back.   Can't we just call my_simhealthcheck from my_checkpoint?

Reply:
> These examples need to keep `my_checkpoint` as the call-back. Can't we just call my_simhealthcheck from my_checkpoint?

https://github.com/illinois-ceesd/mirgecom/blob/9cbb9883d33069e6ab34778cc90045db651ab700/examples/vortex-mpi.py#L197

Reply:
Thanks, I missed that!   Typically, users want to perform the health check at checkpoint time (i.e. before stepping occurs).  We don't want to step if we have a bad state. We want to dump, and then stop.  So I would still add health check as a utility that I call from checkpoint.

Reply:
@MTCam The only reason I put the health check as a `post_step_callback` was to exercise that branch in the stepper :laughing: But I totally get what you're saying. Would you prefer that I merge these into a single callback?

Reply:
I agree with and completely sympathize with why you put it as a post-call.  In my opinion, we should make the examples in the repo look like what we want the users to do.  Users will almost 100% just copy existing example drivers and tweak them. 

If we craft the examples in a way that makes the most sense for production drivers, then we'll automatically get that practice into the DNA of the simulations we need to support.  

The post hook could be exercised by something that looks more like what we'd expect the users to use it for.  Post-step filtering would probably work perfect here.